src/coreclr/jit/importer.cpp

```diff
@@ -10883,6 +10883,22 @@ bool Compiler::impReturnInstruction(int prefixFlags, OPCODE& opcode)
         }
     }
 
+#ifdef SWIFT_SUPPORT
+    // If this method has a SwiftError* out parameter, we need to set the error register upon returning.
+    // By placing the GT_SWIFT_ERROR_RET node before the GT_RETURN node,
+    // we don't have to teach the JIT that GT_SWIFT_ERROR_RET is a valid terminator for BBJ_RETURN blocks.
+    // During lowering, we will move the GT_SWIFT_ERROR_RET to the end of the block
+    // so the error register won't get trashed when returning the normal value.
+    if (lvaSwiftErrorArg != BAD_VAR_NUM)
+    {
+        assert(info.compCallConv == CorInfoCallConvExtension::Swift);
+        assert(lvaSwiftErrorLocal != BAD_VAR_NUM);
+        GenTree* const swiftErrorNode = gtNewLclFldNode(lvaSwiftErrorLocal, TYP_I_IMPL, 0);
+        op1                           = new (this, GT_SWIFT_ERROR_RET)
+            GenTreeOp(GT_SWIFT_ERROR_RET, op1->TypeGet(), swiftErrorNode, op1->gtGetOp1());
+    }
```

Comment:
The comment is outdated now, right?

Reply:
You can use `op1->SetOper(GT_SWIFT_ERROR_RET)` instead here to avoid the new node allocation. Of course that requires some shuffling of the operands, so just allocating a new node is also ok with me, if you like that better feel free to ignore this.